### PR TITLE
documenting tokens

### PR DESCRIPTION
**Pull request by pwright**

### CODE REVIEW

File: modules/overview/pages/tokens.adoc [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+If both sites are equally accessible, for example two public clouds, then it not important where the token is created.
+However, when using a token, the site you are linking to must be accessible from the site you are linking from.
+For example, if you are creating a {service-network} using both a public and private cluster, you must create the token on the public cluster and use the token from the private cluster.
+
+There are two types of Skupper token:
+
+Claim token (default):: 
++
+--
```
Comment (grs) [inline modules/overview/pages/tokens.adoc:24]: Maybe a note here to clarify that a claim token will be automatically exchanged for a certificate by the linking site, preventing the certificate from needing to be exposed at all.

File: modules/overview/pages/tokens.adoc [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+* time - preventing reuse after a specified period.
+* usage - preventing multiple links being created from a single token.
+--
+
+Cert token ::
++
+--
+A cert token can be reused to create many links from many sites.
+--
```
Comment (grs) [inline modules/overview/pages/tokens.adoc:35]: Again, perhaps clarify that this is directly getting the certificate that would be automatically retrieved for a claim. The certificate is a more sensitive artefact and access to it should be carefully restricted. (e.g. don't leave them lying about).

File: modules/overview/pages/tokens.adoc [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+.Additional information
+
+* See the {CliLink} for information on using the token to create links.
+
+// Type: procedure
+[id='creating-claim-tokens-{context}']
+== Creating cert tokens
+
+[role="system:abstract"]
+Creating cert tokens allows you create many links to the {service-network} from different sites without restrictions.
```
Comment (grs) [inline modules/overview/pages/tokens.adoc:73]: I would reword this. A cert token can be used many times, without restriction. (You can have many links when using claims, but - by default - a single claim token can only be used for one link).

File: modules/overview/pages/tokens.adoc [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+. Create a cert token:
++
+--
+[source,bash,options="nowrap"]
+----
+$ skupper token create $HOME/secret.yaml -t cert
+----
+
+NOTE: Cert tokens are always valid, and can be reused indefinitely.
+--
```
Comment (grs) [inline modules/overview/pages/tokens.adoc:90]: There is now the ability to revoke-access on a site which invalidates any previously issued token of any kind.

Comment (pwright) [reply modules/overview/pages/tokens.adoc:98]: added link

File: modules/overview/pages/tokens.adoc (hunk 1 above)
Comment (pwright) [reply modules/overview/pages/tokens.adoc:24]: added  text

File: modules/overview/pages/tokens.adoc (hunk 2 above)
Comment (pwright) [reply modules/overview/pages/tokens.adoc:43]: added text

File: modules/overview/pages/tokens.adoc (hunk 3 above)
Comment (pwright) [reply modules/overview/pages/tokens.adoc]: Not sure what to do here, reworded but this sentence `abstract` is specially targeted to SEO, and should explain the topic, so 'can be used many times' is a bit vague. Would it help if I later 'recommended' users to use claim tokens?

Comment (grs) [reply modules/overview/pages/tokens.adoc]: My point here is that as currently worded it sounds as if there might be some restriction on the number of links from a site when using claims. That is not the case. The only (potential) restriction is that a *given* claim may be used only once (or however many times it was configured to be valid for).

I do think that using claims is generally recommended (which is why we changed the default). That is a somewhat separate point though.

File: modules/overview/pages/tokens.adoc [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+:context: skupper-tokens
+//Category: skupper-tokens
+// Type: assembly
+[id='using-skupper-tokens-{context}']
+
+= Using Skupper tokens
+
+[role="system:abstract"]
+Skupper tokens allow you create links between sites.
```
Comment (briandooley) [inline modules/overview/pages/tokens.adoc:11]:
```suggestion
Skupper tokens allow you to create links between sites.
```

File: modules/overview/pages/tokens.adoc [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+You create a token on one site and use that token from the other site to create a link between the two sites.
+
+NOTE: Although the linking process is directional, a Skupper link allows communication in both directions.
+
+If both sites are equally accessible, for example two public clouds, then it not important where the token is created.
```
Comment (briandooley) [inline modules/overview/pages/tokens.adoc:16]:
```suggestion
If both sites are equally accessible, for example, two public clouds, then it is not important where you create the token.
```
"is created" - passive voice.

File: modules/overview/pages/tokens.adoc [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+However, when using a token, the site you are linking to must be accessible from the site you are linking from.
```
Comment (briandooley) [inline modules/overview/pages/tokens.adoc:17]:
```suggestion
However, when using a token, the site you link to must be accessible from the site you link from.
```

File: modules/overview/pages/tokens.adoc [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+
+There are two types of Skupper token:
+
+Claim token (default):: 
++
+--
+A claim token can be restricted by:
+
+* time - preventing reuse after a specified period.
+* usage - preventing multiple links being created from a single token.
```
Comment (briandooley) [inline modules/overview/pages/tokens.adoc:28]:
```suggestion
* usage - preventing multiple links from being created from a single token.
```

File: modules/overview/pages/tokens.adoc [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+
+All inter-site traffic is protected by mutual TLS using a private, dedicated certificate authority (CA).
+A claim token is not a certificate, but is securely exchanged for a certificate during the linking process.
+By implementing appropriate restrictions (for example, creating a single use claim token) you can avoid the accidental exposure of certificates.
```
Comment (briandooley) [inline modules/overview/pages/tokens.adoc:32]:
```suggestion
By implementing appropriate restrictions (for example, creating a single-use claim token), you can avoid the accidental exposure of certificates.
```

File: modules/overview/pages/tokens.adoc [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+
+. Create a claim token, for example:
++
+--
+[source,bash,options="nowrap"]
+----
+$ skupper token create $HOME/secret.yaml --expiry 30m0s --uses 2 -t claim
+----
+
+NOTE: Claim tokens are the default, the `-t claim` section of the command is unnecessary.
```
Comment (briandooley) [inline modules/overview/pages/tokens.adoc:66]:
```suggestion
NOTE: Claim tokens are the default,. The `-t claim` section of the command is unnecessary.
```

File: modules/overview/pages/tokens.adoc [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+.Additional information
+
+* See the {CliLink} for information on using the token to create links.
+
+// Type: procedure
+[id='creating-claim-tokens-{context}']
+== Creating cert tokens
+
+[role="system:abstract"]
+A cert token allows you create many links to the {service-network} from different sites without restrictions.
```
Comment (briandooley) [inline modules/overview/pages/tokens.adoc:81]:
```suggestion
You can use a cert token to create many links to the {service-network} from different sites without restrictions.
```
"allows you" - Anthropomorphism.

File: modules/overview/pages/tokens.adoc [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+
+. Create a cert token:
++
+--
+[source,bash,options="nowrap"]
+----
+$ skupper token create $HOME/secret.yaml -t cert
+----
+
+NOTE: Cert tokens are always valid, and can be reused indefinitely unless revoked as described in xref:revoking-access-{context}[]
```
Comment (briandooley) [inline modules/overview/pages/tokens.adoc:97]:
```suggestion
NOTE: Cert tokens are always valid and can be reused indefinitely unless revoked as described in xref:revoking-access-{context}[]
```

File: modules/overview/pages/tokens.adoc [hunk 13]
```diff
@@ -0,0 +1,162 @@
...
+A cert token is a certificate issued by the dedicated CA.
+Protect it appropriately.
+--
+
+// Type: procedure
+[id='creating-claim-tokens-{context}']
+== Creating claim tokens
+
+[role="system:abstract"]
+Creating claim tokens allows you to restrict access to the {service-network}
```
Comment (briandooley) [inline modules/overview/pages/tokens.adoc:50]:
```suggestion
You can use claim tokens to restrict access to the {service-network}
```
Anthropmorphism.

File: modules/overview/pages/tokens.adoc [hunk 14]
```diff
@@ -0,0 +1,162 @@
...
+* See the {CliLink} for information on using the token to create links.
+
+// Type: procedure
+[id='revoking-access-{context}']
+== Revoking access to a site
+
+[role="system:abstract"]
+To secure access to the {service-network} you can invalidate all the tokens created from a site.
+
+This option removes all links to the site, and requires that you recreate any links to restore the {service-network}.
```
Comment (briandooley) [inline modules/overview/pages/tokens.adoc:111]:
```suggestion
This option removes all links to the site and requires that you recreate any links to restore the {service-network}.
```

File: modules/overview/pages/tokens.adoc [hunk 15]
```diff
@@ -0,0 +1,162 @@
...
+--
+[source,bash,options="nowrap"]
+----
+$ skupper status
+Skupper is enabled for namespace "west" in interior mode. It is linked to 1 other site.
+$ skupper link status
+Link link1 is active
+----
+
+This shows that incoming links are revoked, but outgoing links are still active.
```
Comment (briandooley) [inline modules/overview/pages/tokens.adoc:155]:
```suggestion
The output shows that the `skupper revoke-access` command has revoked the incoming links, but outgoing links are still active.
```
Defined what "This shows' is referring to and removed passive voice.

File: modules/overview/pages/tokens.adoc [hunk 16]
```diff
@@ -0,0 +1,162 @@
...
+
+You can remove that link using the `skupper link delete link1` command, but to revoke access you must follow this procedure while logged into the appropriate cluster.
```
Comment (briandooley) [inline modules/overview/pages/tokens.adoc:157]:
```suggestion
You can remove that link using the `skupper link delete link1` command, but to revoke access, you must follow this procedure while logged into the appropriate cluster.
```

**Review by briandooley — CHANGES_REQUESTED** [review]

File: modules/overview/pages/tokens.adoc (hunk 8 above)
Comment (pwright) [reply modules/overview/pages/tokens.adoc:28]: will reword, the gerund makes both  bullets difficult to read

File: modules/overview/pages/tokens.adoc (hunk 3 above)
Comment (pwright) [reply modules/overview/pages/tokens.adoc]: @grs, got it, thx

File: modules/overview/pages/tokens.adoc [hunk 17]
```diff
@@ -0,0 +1,164 @@
...
+.Additional information
+
+* See the {CliLink} for information on using the token to create links.
+
+// Type: procedure
+[id='revoking-access-{context}']
+== Revoking access to a site
+
+[role="system:abstract"]
+To secure access to the {service-network} you can invalidate all the tokens created from a site.
```
Comment (grs) [inline modules/overview/pages/tokens.adoc:111]: Rtaher than 'to secure access' I would say, something like 'should a token by compromised, you can prevent unauthorised use of that token by invalidating all tokens'
